core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -1017,8 +1017,15 @@ private[zio] final class FiberContext[E, A](
   }
 
   private[this] def reportUnhandled(v: Exit[E, A], trace: ZTraceElement): Unit = v match {
-    case Exit.Failure(cause) => log(() => cause.prettyPrint, ZIO.someDebug, trace = trace)
-    case _                   =>
+    case Exit.Failure(cause) =>
+      try {
+        log(() => cause.prettyPrint, ZIO.someDebug, trace = trace)
+      } catch {
+        case t: Throwable =>
```

Comment:
Should probably only catch "non-fatal" exceptions as defined by `runtimeConfig`.

Reply:
Yes, as long as that doesn't itself throw an exception.